src/semiwrap/tool/update_yaml.py

```diff
@@ -0,0 +1,283 @@
+import os
+import pathlib
+import sys
+import traceback
+import tempfile
+import typing as T
+import shutil
+import dictdiffer
+import dictdiffer.utils
+from ruamel.yaml import YAML
+
+from ..autowrap.generator_data import MissingReporter
+from ..cmd.header2dat import make_argparser, generate_wrapper
+from ..makeplan import InputFile, makeplan, BuildTarget, CompilerInfo
+
+
+class YamlUpdater:
+    """
+    This class will parse the headers for a robotpy project and create or update their corresponding yaml files.
+
+    The process for doing this goes as such:
+        1. Copy backups of the yaml files into a temporary directory
+        2. Make a temporary directory where the headers will be parsed from scratch
+        3. Merge the additions / deletions from the fresh generation with the backups, and place them into the desired output folder
+
+    It is made to handle custom output directories, but by default it will behave as if it was run "in place"
+
+    This process will keep hand edited values like "ignore", "extra_includes", etc, but should be smart enough to delete config
+    items that no longer exist.
```

Comment:
The prior tool did a dry run by default, that would be preferable -- especially if it's going to delete files.

It would be even better if the dry run output unified diffs instead of what it currently does? ... but that's probably done during parsing and would be annoying to change.